internal/dataplane/parser/translate_ingress.go

```diff
@@ -206,23 +206,18 @@ func (p *Parser) ingressRulesFromIngressV1() ingressRules {
 			&ingressList[j].CreationTimestamp)
 	})
 
-	servicesCache := make(kongServicesCache)
+	// Collect all default backends and TLS SNIs.
 	var allDefaultBackends []netv1.Ingress
 	for _, ingress := range ingressList {
 		ingressSpec := ingress.Spec
-
 		if ingressSpec.DefaultBackend != nil {
 			allDefaultBackends = append(allDefaultBackends, *ingress)
 		}
-
 		result.SecretNameToSNIs.addFromIngressV1TLS(ingressSpec.TLS, ingress)
-
-		if wasServicesCacheUpdated := p.ingressV1ToKongService(ingress, icp, servicesCache); wasServicesCacheUpdated {
-			p.reportKubernetesObjectUpdate(ingress)
-		}
 	}
 
-	for _, service := range servicesCache {
+	// Translate Ingress objects into Kong Services.
+	for _, service := range p.ingressesV1ToKongServices(ingressList, icp) {
```

Comment:
Can we add a comment to tell that the parent could be non-unique, but they should be in the same namespace because ingress can only refer to services in the same namespace?

Reply:
I'm not sure if I understand your comment correctly: should I put a comment explaining the `service.Parent` uniqueness and its namespace association just above the `result.ServiceNameToParent[*service.Name] = service.Parent` line? 🤔 IMO it's not the best place to put that, as the parent assignment originally happens somewhere else, here it's just "repacked" to the result. How about adding a comment above `Parent` field in the `kongstate.Service` struct? a6f9a53